crates/sui-core/src/consensus_handler.rs

```diff
@@ -357,6 +370,22 @@ impl<T: ParentSync + Send + Sync> ExecutionState for ConsensusHandler<T> {
     }
 }
 
+fn reorder_by_gas_price(sequenced_transactions: &mut [VerifiedSequencedConsensusTransaction]) {
+    sequenced_transactions.sort_by_key(|txn| {
+        // Reverse order, so that transactions with higher gas price are put to the beginning.
+        std::cmp::Reverse({
+            match &txn.0.transaction {
+                SequencedConsensusTransactionKind::External(ConsensusTransaction {
+                    tracking_id: _,
+                    kind: ConsensusTransactionKind::UserTransaction(cert),
+                }) => cert.gas_price(),
+                // Non-user transactions are considered to have gas price of zero and are put to the end.
```

Comment:
i believe this sorts ConsensusCommitPrologue txns to the end, which is definitely not what we want.